llama-index-integrations/vector_stores/llama-index-vector-stores-opensearch/llama_index/vector_stores/opensearch/base.py

```diff
@@ -328,16 +333,25 @@ def _knn_search_query(
             search_query = self._default_approximate_search_query(
                 query_embedding, k, vector_field=embedding_field
             )
+        elif self.is_aoss:
+            # if is_aoss is set we are using Opensearch Serverless AWS offering which cannot use
+            # painless scripting so default scoring script returned will be just normal knn_score script
+            search_query = self._default_scoring_script_query(
+                query_embedding,
+                k,
+                space_type=self.space_type,
+                pre_filter={"bool": {"filter": pre_filter}},
+                vector_field=embedding_field,
+            )
         else:
             # https://opensearch.org/docs/latest/search-plugins/knn/painless-functions/
-            search_query = self._default_painless_scripting_query(
+            search_query = self._default_scoring_script_query(
                 query_embedding,
                 k,
                 space_type="l2Squared",
```

Comment:
Shouldn't this match `self.space_type` or nah?

Reply:
Yeah I wasnt totally sure if that was best here. since the last set of changes used painless scritping and were using space type to hold the l2squared function, which is a variant of the l2 distance metric in the non painless score script, I thought to just keep it defaulted to that. I think it might have been hard coded before but let me double check.

Reply:
basically l2squared I am not sure depends on the exact space or not, might need to read up on that more.

Reply:
It seems like it was hardcoded before, so probably fine to leave for now. Just double checking.

Can always improve in a follow-up PR

Reply:
yeah totally. If others have an idea how best we should proceed on the painless scripting def open to ideas as well.